File: DI_IOC_Observer/DI_IOC_Observer_JAVA/src/com/kretzer/PolygonInterface.java
package com.kretzer;

import com.sun.javaws.exceptions.InvalidArgumentException;

/**
 * Created by jason for DI_IOC_Observer_JAVA.
 */
public interface PolygonInterface {
    public double getPerimeter();
    public double getSideLength();
    public void setSideLength(double newSideLength) throws InvalidArgumentException;
    public int getNumberOfSides();
}

File: DI_IOC_Observer/DI_IOC_Observer_CS/DI_IOC_Observer_CS/PolygonInterface.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DI_IOC_Observer_CS
{
    public interface PolygonInterface
    {
        double getPerimeter();
        double getSideLength();
        void setSideLength(double newSideLength);
        int getNumberOfSides();
    }
}